app/views/unattended/provisioning_templates/snippet/_kickstart_ifcfg_generic_interface.erb

```diff
@@ -40,7 +40,7 @@ DEFROUTE=<%= primary %>
 <%-     end -%>
 <%-   end -%>
 <%- end -%>
-<%- if @interface.virtual? && ((!@subnet.nil? && @subnet.has_vlanid?) || @interface.vlanid.present?) -%>
+<%- if @interface.virtual? && ((!@subnet.nil? && @subnet.has_vlanid?) || (@interface.tag.present? && @interface.vlanid.present?)) -%>
```

Comment:
Because if `@subnet.nil? #=> true` and `@interface.tag.blank? #=> true`, then you get [undefined method 'vlanid' for nil](https://github.com/theforeman/foreman/blob/develop/app/models/nic/interface.rb#L30)